### ISSUE TITLE
Guest can see spaces with visibility "Members Only" in directory

### ISSUE THREAD

**Issue by buddh4** [humhub/humhub#3709]
A guest user can see, but not access spaces with visibility "Logged in Users Only"

**Comment by Ace2213** [comment]
I think this is related to #3679. It would be practical to create a visibility directive that controls visibility in the directory, the space picker and other places (like the new/most active spaces widget).

**Comment by donni106** [comment]
Does this relate to being able accessing /directory/members as a guest seeing users listed that are registered, while the profiles itself are behind a login?

**Comment by luke-** [comment]
The users are displayed in Spaces/Directory, but more detailed profile infos like About are not accessible for guests.
So it is not planned to "hide" such users from guests. 

Do you think we should change this behavior?

**Comment by donni106** [comment]
I checked a bit. Private spaces cannot be accessed by guests, so the list of members cannot be seen.
But as mentioned by @buddh4 I was also able to see a private space in the "New spaces" widget in the sidebar:

![Bildschirmfoto 2022-06-28 um 10 50 49](https://user-images.githubusercontent.com/1942953/176137024-a4e6cee7-0246-4bdb-ab45-d5fb4bb198fd.png)

The space with "T" is not listed in the directory, thats correct, but shown in the widget, which I assume false.

**Comment by donni106** [comment]
In addition it shouldn't be able to see members in a private space when clicking on the number of members here:

<img width="471" alt="Bildschirmfoto 2022-06-28 um 11 05 04" src="https://user-images.githubusercontent.com/1942953/176140484-5e7f1c4e-dec4-4105-b3a3-29d60132de2e.png">

I could not find a setting to avoid the modal to pop up:

<img width="891" alt="Bildschirmfoto 2022-06-28 um 11 05 20" src="https://user-images.githubusercontent.com/1942953/176140613-52d616f5-93e2-4061-ba13-54ba682ff0ee.png">

Is there a way to achieve that or find an idea to implement so?

**Comment by luke-** [comment]
There is also a Members section in the right sidebar.

**Comment by donni106** [comment]
> There is also a Members section in the right sidebar.

That can be hidden per space setting:

<img width="464" alt="Bildschirmfoto 2022-06-28 um 17 52 24" src="https://user-images.githubusercontent.com/1942953/176224775-be339c55-1651-4d0f-a83f-b4201fc4ff0e.png">

(_For others, its German for: "Hide member overview in stream"_)

**Comment by dantefromhell** [comment]
> In addition it shouldn't be able to see members in a private space when clicking on the number of members

Yes I tripped over this too. My assumption was that a private Space does not reveal information about the amount of members, the actual members or about it's existence unless you know the exact URL of the space - so one can request becoming a member.

Although I can also see how this could be a per space configuration option so one can completly hide a space from non-members and make it truely invite-only.

Information about Space member count, the members list, activities, ... would only be available to Space members - never non-members in all modules/ widgets.
